src/plugins/intel_gpu/src/graph/reshape.cpp

```diff
@@ -164,7 +164,7 @@ std::string reshape_inst::to_string(reshape_node const& node) {
 
 reshape_inst::typed_primitive_inst(network& network, reshape_node const& node) :
         parent(network, node, (!node.can_be_optimized() && node.get_output_layout().is_static()) ? true : false) {
-    auto input_layout = node.input().get_output_layout();
+    auto input_layout = node.input().get_output_layout(true, node.get_dependency_with_port(0).second);
```

Comment:
I think `node.get_input_layout()` can be used instead

Reply:
Though `get_input_layout` impl may need similar fix. Looks like `get_input_layouts()` respects out port, but `get_input_layout` doesn't.